Libraries/Image/RCTImageLoader.m

```diff
@@ -74,8 +74,7 @@ + (void)loadImageWithTag:(NSString *)imageTag callback:(void (^)(NSError *error,
           // doesn't spike the memory up during the process.
           @autoreleasepool {
             ALAssetRepresentation *representation = [asset defaultRepresentation];
-            ALAssetOrientation orientation = [representation orientation];
```

Comment:
Since this is blocking you I would fork React Native and publish the app with the patched version. Since this commit is small it should be pretty easy to rebase on top of new updates to React Native until the camera roll and image resource handling is improved.